src/schema/sources.ts

```diff
@@ -1256,5 +1262,48 @@ export const resolvers: IResolvers<any, Context> = {
   }),
   Source: {
     permalink: (source: GQLSource): string => getSourceLink(source),
+    referralUrl: async (source: GQLSource, _, ctx): Promise<string> => {
+      if (!ctx.userId) {
+        return null;
+      }
+
+      if (source.type !== SourceType.Squad) {
+        return null;
+      }
+
+      const referralUrl = new URL(
+        `/squads/${source.handle}`,
+        process.env.COMMENTS_PREFIX,
+      );
+
+      if (source.public) {
+        referralUrl.searchParams.append('cid', 'squad');
+        referralUrl.searchParams.append('userid', ctx.userId);
+      } else {
+        let referralToken = null;
+
+        if (!referralToken) {
+          const sourceMember: Pick<SourceMember, 'referralToken'> =
+            await ctx.con.getRepository(SourceMember).findOne({
+              select: ['referralToken'],
+              where: { sourceId: source.id, userId: ctx.userId },
+            });
+
+          referralToken = sourceMember?.referralToken;
+        }
+
+        if (!referralToken) {
+          return null;
+        }
```

Comment:
Wandering if there is a better way to request this in terms of our graphql resolver. 

My logic is that of course `currentMember` might not be available in query context so I have to refetch but it feels clunky.

Also should I just use new `DataLoaderService` to have the whole `referralUrl` cached in request.

Reply:
If `currentMember` is available in the `source` parameter we should use it. It's currently missing in your implementation. Another option is to move `referralUrl` to be inside the `SourceMember` object instead of `Source`. It will make the implementation a bit nicer I believe, but up to you to decide.

As for the second question I'm not sure I understand. Can you elaborate please?

Also, have you checked that the dataloader works in terms of reducing the number of requests?

Reply:
> Other way around it would need to source.handle which is again not guaranteed.

You're right. So let's keep it in `source` but just add a check to see if it was already requested. Please make sure that it's indeed doesn't generate another query.

> Also should I just use new DataLoaderService to have the whole referralUrl cached in request.
Yes. It's better this way I believe. Most probably it will rarely be helpful but better have it there